### ISSUE TITLE
selector-no-type throwing errors when using :nth-child and keyframes

### ISSUE THREAD

**Issue by danielgamble** [stylelint/stylelint#609]
I recreated it here: https://github.com/danielgamble/styletest

Just needs an `npm install`

**Comment by keithamus** [comment]
I've also just noticed `selector-no-type` fails for keyframes as well:

```css
@keyframes spin {
  0% { /* selector-no-type complains here */
    transform: rotate(0deg);
  }
  100% { /* selector-no-type complains here */
    transform: rotate(360deg);
  }
}
```

**Comment by danielgamble** [comment]
I think i was getting that yesterday too but because i removed my keyframes it didn't persist for me, i'll add it to my test repo quickly

**Comment by keithamus** [comment]
:+1: thanks

**Comment by jeddy3** [comment]
Thanks for reporting. This appears to be related to https://github.com/postcss/postcss-selector-parser/issues/14 and https://github.com/postcss/postcss-selector-parser/issues/14, and so might require workarounds at our end. Ideas?

**Comment by jeddy3** [comment]
For the keyframes can we just skip over the tag selector that contain `%`, or are `from` and `to`? For nth-child can we skip: numeric values, `n` and those that are an integer followed by `n`?

**Comment by jeddy3** [comment]
Fixed in #614

**Comment by jeddy3** [comment]
Released as `3.2.2`.
